### ISSUE TITLE
New rules cause dnsmasq to be restarted instead of reloaded

### ISSUE THREAD

**Issue by rhtyd** [apache/cloudstack#3613]
When adding/configuring new dhcp/dns rules, the dnsmasq process is restarted:
https://github.com/apache/cloudstack/blob/master/systemvm/debian/opt/cloud/bin/cs/CsDhcp.py#L71

Explore if it's possible to simply reload the dnsmasq process than restart it.

(If dnsmasq restarts at the same time monitoring service check kicks in, it might show to recover the dnsmasq again)

##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
VR
~~~

**Comment by onitake** [comment]
Restarting should not be necessary for reloading DNS and DHCP records, according to the man page: http://www.thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html#lbAG
Quote:
> When it receives a SIGHUP, dnsmasq clears its cache and then re-loads /etc/hosts and /etc/ethers and any file given by --dhcp-hostsfile, --dhcp-hostsdir, --dhcp-optsfile, --dhcp-optsdir, --addn-hosts or --hostsdir. The dhcp lease change script is called for all existing DHCP leases. If --no-poll is set SIGHUP also re-reads /etc/resolv.conf. SIGHUP does NOT re-read the configuration file. 

The systemd unit in Debian buster sends a SIGHUP on reload, so it should suffice:
```
ExecReload=/bin/kill -HUP $MAINPID
```

**Comment by andrijapanicsb** [comment]
it WAS necesary some days (years?) ago, since there was no way to reload the dnsmasq (or it was buggy) - but today - you are right.

**Comment by rhtyd** [comment]
This will require large changes, moved to 4.15

**Comment by rhtyd** [comment]
It may be worth discussing and move it out if the scope/changes are huge. cc @nvazquez @weizhouapache

**Comment by pdion891** [comment]
Reloading dnsmasq when creating a new vm would probably work, but when you introduce a new network tier in a VPC, you will probably have to restart dnsmasq so it introduce the new network interface of the VR in dnsmasq configuration.

**Comment by weizhouapache** [comment]
@pdion891 
yes, this is why I keep the following lines in CsDhcp.py in #5319 

```
        if self.conf.commit():
            restart_dnsmasq = True
```

**Comment by weizhouapache** [comment]
fixed by #5319
